### ISSUE TITLE
rationalize `Index.select(...)` return/exception values

### ISSUE THREAD

**Issue by ctb** [sourmash-bio/sourmash#1940]
In https://github.com/sourmash-bio/sourmash/pull/1936, I realized that when confronted with a `select` that will result in an empty `Index` class, some classes (`LCA_Database` in particular) raise a ValueError, while others just return an empty `Index` class.

See `test_index_select_nada` in particular.

I'm not sure what it _should_ do but it would be good to have a single kind of behavior. I think I originally implemented it this way so as to provide good error messages to people when loading the wrong k-mer size for an LCA or SBT database.

**Comment by ctb** [comment]
> I'm not sure what it _should_ do but it would be good to have a single kind of behavior. I think I originally implemented it this way so as to provide good error messages to people when loading the wrong k-mer size for an LCA or SBT database.

I tried changing LCAs and SBTs over to returning an empty database on incompatible select in https://github.com/sourmash-bio/sourmash/pull/2204, but I then reverted that change, because so much useful information was lost in the process!

Instead, once #2204 is merged, we just make an empty database in `sourmash_args.load_dbs_and_sigs` when `select` raises a ValueError.  This leaves the internal API a little messier, but keeps informative error messages around.
